### PR TITLE
50DPS Urinal fix

### PR DESCRIPTION
**Pull request by HouseOfSynth**
Fixes urinals not having armour checks or click cooldowns when agg-grabbing them against them.

They now do a maximum of 8 brute to the head and have a cooldown slightly higher than the default attack cooldown. Armour is now taken into account aswell.

### CODE REVIEW

**Comment by HouseOfSynth** [comment]
!bugfix

**Comment by HouseOfSynth** [comment]
!review

File: code/game/objects/structures/watercloset.dm [hunk 1]
```diff
@@ -109,7 +109,7 @@
 	density = 0
 	anchored = 1
 
-/obj/structure/urinal/attackby(obj/item/I as obj, mob/user as mob)
+/obj/structure/urinal/attackby(obj/item/I as obj, mob/user as mob, var/damtype = BRUTE)
```
Comment (Geevies) [inline code/game/objects/structures/watercloset.dm:112]:
```suggestion
/obj/structure/urinal/attackby(obj/item/I, mob/user)
```

Comment (HouseOfSynth) [reply code/game/objects/structures/watercloset.dm:112]: Oh woops, forgot we defined that in apply_damage()

File: code/modules/mob/living/simple_animal/shade.dm [hunk 2]
```diff
@@ -95,7 +95,7 @@
 	var/datum/weakref/original_body
 	var/datum/weakref/possessed_body
 
-/mob/living/simple_animal/shade/bluespace/apply_damage(var/damage_flags, var/def_zone, var/used_weapon)
+/mob/living/simple_animal/shade/bluespace/apply_damage(var/damage = 0,var/damagetype = BRUTE, var/def_zone = null, var/blocked = 0, var/used_weapon = null, var/damage_flags)
```
Comment (NonQueueingMatt) [inline code/modules/mob/living/simple_animal/shade.dm:98]: this stuff should be null by default:
`/mob/living/simple_animal/shade/bluespace/apply_damage(var/damage, var/damagetype, var/def_zone, var/blocked, var/used_weapon, var/damage_flags)`

Comment (HouseOfSynth) [reply code/modules/mob/living/simple_animal/shade.dm]: Fair enough, thanks!

File: code/modules/mob/living/simple_animal/shade.dm [hunk 3]
```diff
@@ -95,7 +95,7 @@
 	var/datum/weakref/original_body
 	var/datum/weakref/possessed_body
 
-/mob/living/simple_animal/shade/bluespace/apply_damage(var/damage_flags, var/def_zone, var/used_weapon)
+/mob/living/simple_animal/shade/bluespace/apply_damage(var/damage, var/damagetype, var/def_zone, var/blocked, var/used_weapon, var/damage_flags)
```
Comment (Alberyk) [inline code/modules/mob/living/simple_animal/shade.dm:98]: Why is this here?

Comment (NonQueueingMatt) [reply code/modules/mob/living/simple_animal/shade.dm:98]: fails travis otherwise, doesn't matter
